python_modules/dagster/dagster/_experimental/staging_io_manager.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Optional, Sequence, Union
+
+from dagster import InputContext, OutputContext
+from dagster._core.definitions.events import AssetMaterialization, AssetObservation
+from dagster._core.definitions.metadata import (
+    MetadataEntry,
+    PartitionMetadataEntry,
+    TextMetadataValue,
+)
+from dagster._core.events import AssetObservationData, DagsterEventType, StepMaterializationData
+from dagster._core.events.log import EventLogEntry
+from dagster._core.storage.io_manager import IOManager
+
+
+def get_asset_materialization_from_event_log_entry(
+    event_log_entry: EventLogEntry,
+) -> Optional[AssetMaterialization]:
+    if event_log_entry.dagster_event_type != DagsterEventType.ASSET_MATERIALIZATION:
+        return None
+    dagster_event = event_log_entry.dagster_event
+    if not dagster_event:
+        return None
+    if not isinstance(dagster_event.event_specific_data, StepMaterializationData):
+        return None
+
+    return (
+        dagster_event.event_specific_data.materialization
+        if isinstance(dagster_event.event_specific_data.materialization, AssetMaterialization)
+        else None
+    )
+
+
+def get_asset_observation_from_event_log_entry(
+    event_log_entry: EventLogEntry,
+) -> Optional[AssetObservation]:
+    if event_log_entry.dagster_event_type != DagsterEventType.ASSET_OBSERVATION:
+        return None
+    dagster_event = event_log_entry.dagster_event
+    # assert dagster_event
+    if not dagster_event:
+        return None
+    # assert isinstance(dagster_event.event_specific_data, AssetObservationData)
+    if not isinstance(dagster_event.event_specific_data, AssetObservationData):
+        return None
+
+    return (
+        dagster_event.event_specific_data.asset_observation
+        if isinstance(dagster_event.event_specific_data.asset_observation, AssetObservation)
+        else None
+    )
+
+
+def get_text_metadata(
```

Comment:
`AssetMaterialization` now has a `metadata` property that I believe should make this more straightforward

Reply:
👍🏻